ui/lib/db.ts

```diff
@@ -262,3 +317,155 @@ export async function getNextUnacceptedEntryKey(dataset: string): Promise<string
     .executeTakeFirst();
   return row?.key ?? null;
 }
+
+export interface IPositionFilters {
+  dataset?: string;
+  entity_id?: string;
+  caption?: string;
+  country?: string;
+  is_pep?: boolean | null;
+  q?: string;
+  sort?: string[];
+  limit?: number;
+  offset?: number;
+}
+
+export interface IPositionListResponse {
+  limit: number;
+  offset: number;
+  total: number;
+  results: Position[];
+}
+
+export async function getPositionsDatasets(): Promise<string[]> {
+  const db = await getDb();
+
+  const rows = await db
+    .selectFrom(POSITION_TABLE_NAME)
+    .select('dataset')
+    .where('deleted_at', 'is', null)
+    .distinct()
+    .orderBy('dataset', 'asc')
+    .execute();
+
+  return rows.map(row => row.dataset);
+}
+
+export async function getPositions(filters: IPositionFilters = {}): Promise<IPositionListResponse> {
+  const {
+    dataset,
+    entity_id,
+    caption,
+    country,
+    is_pep,
+    q,
+    limit = 50,
+    offset = 0,
+  } = filters;
+
+  const db = await getDb();
+
+  // Build the base query
+  let query = db
+    .selectFrom(POSITION_TABLE_NAME)
+    .where('deleted_at', 'is', null);
+
+  if (entity_id !== undefined) {
+    query = query.where('entity_id', '=', entity_id);
+  }
+  if (caption !== undefined) {
+    query = query.where('caption', '=', caption);
+  }
+  if (country !== undefined) {
+    // TODO(Leon Handreke): How safe is this? I don't know.
```

Comment:
it's parameterised :) but good question

<img width="919" height="54" alt="image" src="https://github.com/user-attachments/assets/cb2129eb-5723-46c8-8de5-46059ba07b20" />